### ISSUE TITLE
Enhancement:  add ability to assign user to role on a project

### ISSUE THREAD

**Issue by lhasadad** [terraform-providers/terraform-provider-openstack#237]
Suggestion for additional capabilities. 

One item I am currently unable to do with the openstack provider is to map a user to a role on a project. I can create a project with the provider, but I need to manually add a user with a role to the project before I can continue using terraform to populate the project.   I would be better to be able to do this as one apply action.

### Terraform Version
$ ./terraform.exe -v
Terraform v0.11.3
+ provider.openstack v1.2.0

### Affected Resource(s)
Projects/Users 
If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
Not a bug  so no config file needed I believe.

### Expected Behavior
What should have happened?
I should have a way through the .tf definition to specify one or more users that should be added with appropriate/specified roles if using a provider connection with admin permissions. 

### Actual Behavior
unable to specify users to add to role on project.

**Comment by jtopjian** [comment]
@lhasadad Thanks for opening this. An `openstack_identity_user_role_v3` resource should be pretty easy to create. I think I can work on this pretty soon.

**Comment by gonzolino** [comment]
Since I need some identity resources for a project, among them `openstack_identity_user_role_v3`, I started working on this over the weekend and created PR #248.

**Comment by jtopjian** [comment]
@lhasadad That might be possible to do but it might be tricky. It sounds like you want to set up multiple `provider` blocks/aliases, each with a different auth scope.

I _think_ this might be possible if you define the `provider` blocks in separate modules.

**Comment by lhasadad** [comment]
@jtopjian Joe,  I have a current set of TF files with module for the project contents.  right now,  I am commenting out the stuff other than the project creation, doing an apply,  then adding the user to the project then uncommenting the project plan and provider alias with the user credentials and running the plan again.   this results in a fully constructed deployment the way I was hoping.  right now the issue is getting the user defined on the fly in the plan.   not sure if this is or is not a possibility with whats planned but would be nice to be able to add the user role assignment resource and have that apply after the projects are created and before the provider for the project creation and the module based on it start to do their thing.

**Comment by jtopjian** [comment]
>  right now the issue is getting the user defined on the fly in the plan.

Right, and this is the tricky part. I admit that I've never done this so I can't be completely confident if it will work or not. I know that Terraform can now instantiate provider blocks within modules using input to the module, so that's why I'm thinking if you have a module with the per-user resource creation, you might be able to pass in the new user scope information and have it work.

But again, this kind of workflow isn't specific to the OpenStack provider per-se. The resources that @gonzolino is working on will certainly provide the admin user administration you're looking for. It's now a matter of getting Terraform to work with the multi-scope, multi-step workflow you want to do.

**Comment by lhasadad** [comment]
So looks like PR 250 hit with this update.  @gonzolino  thank you so much for working on this.  looking forward to giving it a try when it gets dropped into a release.

**Comment by jbcraig** [comment]
I have been testing the use of role assignments and it is working well for me.  One thing I need is the ability to mark the role as --inherited but this doesn't appear to be available.  I have no experience with go so I can't tell if this is an oversight or if it's not available in the gophercloud library.

**Comment by jtopjian** [comment]
@jbcraig Glad to hear it's working :)

I'm not familiar with inherited support. Would this be it? https://developer.openstack.org/api-ref/identity/v3/#os-inherit

If so, it's not available in Gophercloud at the moment.

**Comment by jbcraig** [comment]
Yes, that would be it. Thanks for letting me know. I’ll just need to drop into shell to create those roles.

**Comment by lhasadad** [comment]
@gonzolino I wanted to add my thanks.  we are using this daily to provision projects in openstack.  thanks again.
